File: src/main/java/com/example/jobs/dto/RegisterDeveloperRequest.java
package com.example.jobs.dto;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.time.ZonedDateTime;
import java.util.Collections;
import java.util.List;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@Builder
@JsonIgnoreProperties(ignoreUnknown = true)
public class RegisterDeveloperRequest {
  @JsonProperty("id")
  private UUID id;

  @JsonProperty("email")
  private String email = "";

  @JsonProperty("name")
  private String name = "";

  @JsonProperty("api_key")
  private String apiKey = "";

  @JsonProperty("permissions")
  private List<String> permissions = Collections.emptyList();

  @JsonProperty("rate_limit_per_hour")
  private int rateLimitPerHour = 100;

  @JsonProperty("is_active")
  private boolean isActive = true;

  @JsonProperty("tier")
  private String tier = "free";

  @JsonProperty("created_at")
  private ZonedDateTime createdAt;

  @JsonCreator
  public RegisterDeveloperRequest() {
  }
}

File: src/main/kotlin/com/example/jobs/dto/RegisterDeveloperRequest.kt
package com.example.jobs.dto

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import java.time.ZonedDateTime
import java.util.UUID

@JsonIgnoreProperties(ignoreUnknown = true)
class RegisterDeveloperRequest {
    @field:JsonProperty("id")
    var id: UUID? = null

    @field:JsonProperty("email")
    var email: String = ""

    @field:JsonProperty("name")
    var name: String = ""

    @field:JsonProperty("api_key")
    var apiKey: String = ""

    @field:JsonProperty("permissions")
    var permissions: List<String> = emptyList()

    @field:JsonProperty("rate_limit_per_hour")
    var rateLimitPerHour: Int = 100

    @field:JsonProperty("is_active")
    var isActive: Boolean = true

    @field:JsonProperty("tier")
    var tier: String = "free"

    @field:JsonProperty("created_at")
    var createdAt: ZonedDateTime? = null

    @JsonCreator
    constructor()
}